include/pybind11/detail/internals.h

```diff
@@ -189,6 +189,11 @@ struct internals {
         PYBIND11_TLS_FREE(tstate);
     }
 #endif
+#if PYBIND11_INTERNALS_VERSION >= 5
+    // Note that we have to invoke strdup to allocate memory to ensure a unique address
+    // We want unique addresses since we use pointer equality to compare function records
+    const char *function_record_capsule_name = strdup("pybind11_function_record_capsule");
```

Comment:
> We're leaking this, that's fine.

Yep, all of internals leaks by design. We can never deallocate anything in it.

> We're assuming only one instance is constructed per process.

Incorrect, it's one instance per PYBIND11_INTERNALS_ID. If there are multiple pybind11 modules with different PYBIND11_INTERNALS_IDs, you will get different copies by design.

> Tangential, but since we are making a change here already, I think this would be a good addition, to make the intent clear, and to guard against accidents:

Yep, that's a good idea. Although note that we should also add

> // Always leak, never delete
>  ~internals() = delete;

Reply:
> Incorrect, it's one instance per PYBIND11_INTERNALS_ID. If there are multiple pybind11 modules with different PYBIND11_INTERNALS_IDs, you will get different copies by design.

Oh, of course, thanks for the correction!

> // Always leak, never delete
> ~internals() = delete;

I was curious and tried it out, it fails with a compiler error because there is a destructor already.

Then I added a segfault trigger in the existing destructor (`long *BAD = nullptr; *BAD = 101;`). With that test_embed fails but the pytest tests all pass.

I'm not sure what to make of those observations, but I'm thinking let's not do anything about it in this PR (or anytime soon). I'm not worried at all about the `function_record_capsule_name` leak.

**Tangent**: I was about to write: I know for sure cpython leaks a lot more: https://github.com/rwgk/stuff/blob/f6549ecadb46169feb5f3c2456dcd13852382f7a/noddy/embedded_noddy_main_run.c

But I just found out: it still leaks with Python 3.9, but Python 3.10 is clean!

Anyway, I'm still not worried about the leak.